test-loop-tests/src/utils/peer_manager_actor.rs

```diff
@@ -251,30 +396,41 @@ fn network_message_to_client_handler(
         NetworkRequests::Block { block } => {
             let my_peer_id = shared_state.account_to_peer_id(&my_account_id);
             for account_id in shared_state.accounts() {
-                if account_id != my_account_id {
-                    let future = shared_state
-                        .senders_for_account(&account_id)
-                        .client_sender
-                        .send_async(BlockResponse {
-                            block: block.clone(),
-                            peer_id: my_peer_id.clone(),
-                            was_requested: false,
-                        });
-                    drop(future);
+                if account_id == my_account_id {
+                    continue;
                 }
+
+                let senders = shared_state.senders_for_account(&my_account_id, &account_id);
+
+                let future = senders.client_sender.send_async(BlockResponse {
+                    block: block.clone(),
+                    peer_id: my_peer_id.clone(),
+                    was_requested: false,
+                });
+                drop(future);
+
+                senders.peer_manager_sender.send(TestLoopNetworkBlockInfo {
```

Comment:
Can we try doing this as part of `SetChainInfo` message handler instead of here? That's probably the correct way of doing things as I'm not sure if block got here can be part of forks etc.

Reply:
Good idea, done in [f8605fc](https://github.com/near/nearcore/pull/13299/commits/f8605fc196cc399ce922b8afe68d3d590b64c195)